src/customer_storage_recalculator.py

```diff
@@ -0,0 +1,217 @@
+                    'Name': "TABLE_NAME",
+                    'Value': connection_info["database"]
+                },
+                {
+                    'Name': "APP_VERSION",
+                    'Value': APP_VERSION
+                },
+                {
+                    'Name': "RECALCULATOR_NAME",
+                    'Value': "customer_storage_recalculator"
+                }]
+
+    for key in records["customerChanges"]:
+        logger.info(f"customer delta found - {key}")
+        customer_total_image_size_delta += abs(key["totalsizedelta"])
+        customer_total_image_number_delta += abs(key["numberofimagesdelta"])
+        customer_total_thumbnail_size_delta += abs(key["totalsizeofthumbnailsdelta"])
+
+    metric_data.extend([
+        {
+            'MetricName': CLOUDWATCH_CUSTOMER_IMAGE_SIZE_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': customer_total_image_size_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_CUSTOMER_IMAGE_NUMBER_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': customer_total_image_number_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_CUSTOMER_THUMBNAIL_SIZE_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': customer_total_thumbnail_size_delta
+        }
+    ])
+
+    space_total_image_size_delta = 0
+    space_total_image_number_delta = 0
+    space_total_thumbnail_size_delta = 0
+
+    for key in records["spaceChanges"]:
+        logger.info(f"space delta found - {key}")
+        space_total_image_size_delta += abs(key["totalsizedelta"])
+        space_total_image_number_delta += abs(key["numberofimagesdelta"])
+        space_total_thumbnail_size_delta += abs(key["totalsizeofthumbnailsdelta"])
+
+    metric_data.extend([
+        {
+            'MetricName': CLOUDWATCH_SPACE_IMAGE_SIZE_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': space_total_image_size_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_SPACE_IMAGE_NUMBER_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': space_total_image_number_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_SPACE_THUMBNAIL_SIZE_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': space_total_thumbnail_size_delta
+        }
+    ])
+
+    try:
+        logger.debug(f"updating cloudwatch metrics - {metric_data}")
+        cloudwatch.put_metric_data(MetricData=metric_data,
+                                   Namespace='Entity Counter Recalculator')
+        return metric_data
+    except Exception as e:
+        logger.error(f"Error posting to cloudwatch: {e}")
+        raise e
+
+
+def __run_sql(conn):
```

Comment:
Could you calculate space level changes first then customer level? Would save multiple scans of `"ImageStorage"`